high-level.lisp

```diff
@@ -958,3 +958,44 @@ with upper left block with dimension P-by-Q. Returns the intermediate representa
         (inv (recurse (make-identity-matrix (matrix-rows m)) m (- power)))
         (recurse (make-identity-matrix (matrix-rows m)) m power))))
 
+(define-condition singular-matrix-error (error)
+  ((matrix :initarg :matrix :reader singular-matrix-error-matrix))
+  (:documentation "A singular matrix is used where non-singular expected.")
+  (:report (lambda (condition stream)
+             (format stream "Singular matrix used where non-singular expected~%~A"
+                     (singular-matrix-error-matrix condition)))))
+
+(defun solve (a b)
```

Comment:
Maybe rename it to `linear-solve` in case we want to add non-linear solvers in the future?

In any case, it's likely that this function should be re-purposed later on to take into account the type of matrix that `a` is. For instance, if it is Hermitian you'd want to do a Cholesky factorization instead of LU, etc. Additionally, it would be interesting to dispatch on the scalar type (i.e., complex vs double float).